src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -3678,7 +3678,18 @@ LogicalResult ONNXNonMaxSuppressionOp::inferShapes(
 
 LogicalResult ONNXNonZeroOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  auto builder = mlir::Builder(getContext());
+  Type inputType = getOperand().getType();
+  if (!inputType.isa<RankedTensorType>())
+    return success();
+  SmallVector<int64_t, 2> dims;
+  // The first dimension size is the rank of the input.
```

Comment:
That seems to be the reverse: 1st dim is the number of nonzero... aka -1 here
2nd dim is the rank

``` python
condition = np.array([[1, 0], [1, 1]], dtype=bool)
result = np.array(np.nonzero(condition), dtype=np.int64)  # expected output [[0, 1, 1], [0, 0, 1]]
print(result, result.shape)
```
```
[[0 1 1]
 [0 0 1]] (2, 3)
```

Reply:
No it's not the reverse, as you see (2, 3), 2 is the rank of inputs (2 dimensions) and 3 is the number of nonzeros

Reply:
I lifted the python code from their backend tests... But I agree, it is 3 entries of 2 dim each.